### PR TITLE
Implement delete request and result validating wrappers

### PR DESCRIPTION
**Pull request by iainmcgin**

### CODE REVIEW

**Comment by iainmcgin** [comment]
First step towards implementing #35.

File: protocol/java/org/openyolo/protocol/CredentialDeleteResult.java [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+     * Creates the protocol buffer equivalent to this credential deletion result.
+     */
+    public Protobufs.CredentialDeleteResult toProtobuf() {
+        return Protobufs.CredentialDeleteResult.newBuilder()
+                .setResultCodeValue(mResultCode)
+                .putAllAdditionalProps(mAdditionalProps)
+                .build();
+    }
+
+    public Intent toResultIntentData() {
```
Comment (dxslly) [inline protocol/java/org/openyolo/protocol/CredentialDeleteResult.java]: Missing javadocs from here downwards in this file

File: protocol/javatests/org/openyolo/protocol/CredentialDeleteRequestTest.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            .build();
+
+    @Test
+    public void build() {
+        CredentialDeleteRequest request =
+                new CredentialDeleteRequest.Builder(CREDENTIAL)
+                        .setAdditionalProperties(ADDITIONAL_PROPS)
+                        .build();
+
+        assertThat(request.getCredential()).isNotNull();
```
Comment (dxslly) [inline protocol/javatests/org/openyolo/protocol/CredentialDeleteRequestTest.java]: Replace isNotNull() with isSameAs()

File: protocol/javatests/org/openyolo/protocol/CredentialDeleteRequestTest.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+    @Test
+    public void toProtobuf() {
+        Protobufs.CredentialDeleteRequest proto =
+                new CredentialDeleteRequest.Builder(CREDENTIAL)
+                        .setAdditionalProperties(ADDITIONAL_PROPS)
+                        .build()
+                .toProtobuf();
+
+        assertThat(proto.hasCredential());
```
Comment (dxslly) [inline protocol/javatests/org/openyolo/protocol/CredentialDeleteRequestTest.java]: Missing assertion here

**Pull request merged**